### PR TITLE
Save and load answers for extension bootstraping

### PR DESCRIPTION
**Pull request by bdjilka**

### CODE REVIEW

File: connect/cli/plugins/project/extension/helpers.py [hunk 1]
```diff
@@ -34,8 +37,20 @@ def bootstrap_extension_project(config, output_dir, overwrite):  # noqa: CCR001
...
         grouped_definitions[elem['extension_type']][elem['category']].append(elem)
 
+    questions = get_questions(config, grouped_definitions)
+
+    if load_answers_dir and os.path.splitext(load_answers_dir)[-1] == '.json':
+        with open(load_answers_dir) as fp:
+            loaded_answers = json.load(fp)
+
+        for question in questions:
+            if question['name'] in loaded_answers:
```
Comment (ffaraone) [inline connect/cli/plugins/project/extension/helpers.py:47]: it's a great idea to set the defaults but probably we should open the wizard directly showing the summary page.

File: connect/cli/plugins/project/extension/helpers.py [hunk 2]
```diff
@@ -34,8 +37,20 @@ def bootstrap_extension_project(config, output_dir, overwrite):  # noqa: CCR001
...
+                question['default'] = loaded_answers[question['name']]
+
+        console.secho(f'Answers loaded from {save_answers_dir}', fg='green')
```
Comment (ffaraone) [inline connect/cli/plugins/project/extension/helpers.py:50]: No need IMO to show this message here.

File: connect/cli/plugins/project/extension/helpers.py [hunk 3]
```diff
@@ -60,6 +75,22 @@ def bootstrap_extension_project(config, output_dir, overwrite):  # noqa: CCR001
     if not overwrite and os.path.exists(project_dir):
         raise ClickException(f'The destination directory {project_dir} already exists.')
 
+    if save_answers_dir:
+        if os.path.exists(save_answers_dir):
```
Comment (ffaraone) [inline connect/cli/plugins/project/extension/helpers.py:79]: IMO this check should be done at the very beginning

File: connect/cli/plugins/project/extension/helpers.py [hunk 4]
```diff
@@ -60,6 +75,22 @@ def bootstrap_extension_project(config, output_dir, overwrite):  # noqa: CCR001
...
+            console.secho(
+                f'File {save_answers_dir} already exists. Answers will not be saved.',
+                fg='red',
+            )
+        elif os.path.splitext(save_answers_dir)[-1] != '.json':
```
Comment (ffaraone) [inline connect/cli/plugins/project/extension/helpers.py:84]: I would not check the provided extension

Comment (bdjilka) [reply connect/cli/plugins/project/extension/helpers.py:84]: But in this case user can save it to another format and in future will not be able to use it to load answers. And will not understand why....

File: connect/cli/plugins/project/extension/helpers.py (hunk 3 above)
Comment (bdjilka) [reply connect/cli/plugins/project/extension/helpers.py:79]: Check in the beginning make sense if we need to terminate wizard, either placing of check doesn't matter imo

File: connect/cli/plugins/project/extension/helpers.py (hunk 2 above)
Comment (bdjilka) [reply connect/cli/plugins/project/extension/helpers.py:50]: Ok

File: connect/cli/plugins/project/extension/helpers.py (hunk 1 above)
Comment (bdjilka) [reply connect/cli/plugins/project/extension/helpers.py:47]: No really understand, without ability to change anything in wizard?

File: connect/cli/plugins/project/extension/helpers.py [hunk 5]
```diff
@@ -34,8 +37,20 @@ def bootstrap_extension_project(config, output_dir, overwrite):  # noqa: CCR001
         statuses_by_event[elem['extension_type']][elem['type']] = elem['object_statuses']
         grouped_definitions[elem['extension_type']][elem['category']].append(elem)
 
+    questions = get_questions(config, grouped_definitions)
+
+    if load_answers_dir and os.path.splitext(load_answers_dir)[-1] == '.json':
```
[the remark this answers is not in the document]
Comment (bdjilka) [reply connect/cli/plugins/project/extension/helpers.py:42]: Ok

File: connect/cli/plugins/project/extension/helpers.py (hunk 3 above)
Comment (ffaraone) [reply connect/cli/plugins/project/extension/helpers.py:79]: Yes IMO we should stop if the dest file already exist

File: connect/cli/plugins/project/extension/helpers.py (hunk 4 above)
Comment (ffaraone) [reply connect/cli/plugins/project/extension/helpers.py:84]: But the extension does not guarantee the format

File: connect/cli/plugins/project/extension/helpers.py (hunk 1 above)
Comment (ffaraone) [reply connect/cli/plugins/project/extension/helpers.py:47]: No ,you can always from the summary go back

**Review by ffaraone — APPROVED** [review]

**Pull request merged**